package/MDAnalysis/lib/util.py

```diff
@@ -212,6 +214,182 @@ def filename(name, ext=None, keep=False):
     return name if isstream(name) else str(name)
 
 
+def get_reader_for(filename, format=None):
+    """Return the appropriate trajectory reader class for *filename*.
+
+    Parameters
+    ----------
+    filename
+        filename of the input trajectory or coordinate file
+        Also can handle a few special cases, see notes below.
+
+    Returns
+    -------
+    A Reader object
+
+    Notes
+    -----
+    There are a number of special cases that can be handled:
+      If *filename* is a numpy array, MemoryReader is returned.
+      If *filename* is an MMTF object, MMTFReader is returned.
+      If *filename* is an iterable of filenames, ChainReader is returned.
+
+    Automatic detection is disabled when an explicit *format* is
+    provided.
+    """
+    # Checks for specialised formats
+    # memoryreader slurps numpy arrays
+    if isinstance(filename, np.ndarray):
+        format = 'memory'
+    # mmtf slurps mmtf object
+    elif isinstance(filename, mmtf.MMTFDecoder):
+        format = 'mmtf'
+    # chainreader for iterables
+    # order is important here, np.array is iterable but gets stolen above
+    elif iterable(filename):
+        format = 'chain'
+
+    # else let the guessing begin!
+    if format is None:
+        format = guess_format(filename)
+    format = format.upper()
+    try:
+        return _READERS[format]
+    except KeyError:
+        raise ValueError(
+            "Unknown coordinate trajectory format '{0}' for '{1}'. The FORMATs \n"
+            "           {2}\n"
+            "           are implemented in MDAnalysis.\n"
+            "           See http://docs.mdanalysis.org/documentation_pages/coordinates/init.html#id1\n"
+            "           Use the format keyword to explicitly set the format: 'Universe(...,format=FORMAT)'\n"
+            "           For missing formats, raise an issue at "
+            "http://issues.mdanalysis.org".format(
+                format, filename, _READERS.keys()))
+
+
+def get_writer_for(filename, format=None, multiframe=None):
+    """Return an appropriate trajectory or frame writer class for `filename`.
+
+    The format is determined by the *format* argument or the extension of
+    `filename`. If `format` is provided, it takes precedence over The
+    extension of `filename`.
+
+    Parameters
+    ----------
+    filename : str or ``None``
+        If no *format* is supplied, then the filename for the trajectory is
```

Comment:
I think with Napoleon and numpy formatting, write \``format`\` instead of `*format*`.